### PR TITLE
CI: change np-dev xfails to not strict

### PR DESCRIPTION
**Pull request by simonjayhawkins**
follow-up to #32025

### CODE REVIEW

File: pandas/tests/scalar/timedelta/test_arithmetic.py [hunk 1]
```diff
@@ -386,6 +386,7 @@ def test_td_div_numeric_scalar(self):
                 marks=pytest.mark.xfail(
                     _is_numpy_dev,
                     reason="https://github.com/pandas-dev/pandas/issues/31992",
+                    strict=False,
```
Comment (simonjayhawkins) [inline pandas/tests/scalar/timedelta/test_arithmetic.py]: I _think_ this test is consistently failing. remove the strict=False?

**Comment by jbrockmendel** [comment]
Do we have a good way of knowing when the xfail becomes unnecessary?

**Comment by simonjayhawkins** [comment]
once we make the test xfail(strict=False), i don't think so. still better than a skip though since we'll see the xpass in the test output

**Comment by jreback** [comment]
@jbrockmendel ok with this until have time to diagnose / fix the real cause?

**Comment by jbrockmendel** [comment]
fine by me

**Pull request merged**